app/controllers/admin/api/cms/sections_controller.rb

```diff
@@ -101,9 +102,19 @@ def destroy
   end
 
   private
-    def find_section
-      @section = current_account.sections.find_by_id(params[:id]) || current_account.sections.find_by_system_name(params[:id])
 
-      raise ActiveRecord::RecordNotFound.new("Couldn't find CMS::Section with id or system_name=#{params[:id]}") if @section.nil?
-    end
+  def find_section(contents: false)
+    q = current_account.sections
+    q = q.includes(:pages, :files, :builtins, :children) if contents
+    q = q.where(id: params[:id]).or(q.where(system_name: params[:id]))
+    q.load
+
+    raise ActiveRecord::RecordNotFound, "Couldn't find CMS::Section with id or system_name=#{params[:id]}" if q.empty?
+
+    @section = q.first
```

Comment:
Is `system_name` uniq? I ask because I'm trying to understand if there's a possibility for this query to return multiple rows.
**IF** that's the case and since we only care about the first row, we should add a `limit` to the query. Or even use `find_by` with `OR`... Something like:
```ruby
@section = current_account.sections.tap do |query|
  query.includes(:pages, :files, :builtins, :children) if contents
  query.find_by('id = ? OR system_name = ?', params[:id], params[:id]) # Ouch double params[:id]
end.load
```

Reply:
`system_name` is unique, yes. I prefer to not hardcode any SQL in the code that could make a future rails upgrades harder, so I'll stick to the `.where`  + `.first` approach. However, I actually like the `.tap` styling. I tried to adapt the current query to the tap styling but it doesn't work because `tap` always returns `current_account.sections` and the modifications inside the block are lost. Finally I'm leaving this as it was (sad face).